### ISSUE TITLE
Sharing buttons: Specific sharing configuration results in misleading page editor display

### ISSUE THREAD

**Issue by coder-karen** [Automattic/jetpack#25329]
### Impacted plugin

Jetpack

### Steps to Reproduce

This issue is specific to a particular configuration of sharing settings. While this can be replicated in both beta and previous versions of Jetpack and there are existing, similar issues, this issue is more pronounced as a result of this PR - https://github.com/Automattic/jetpack/pull/24896 - because sharing buttons now shouldn't display at all on pages with the Blog Posts block.

1. On a Jetpack connected site, make sure Sharing and / or Like buttons are enabled via Jetpack > Settings > Sharing.
2. From Settings > Sharing, make sure 'Front Page, Archive Pages, and Search Results' is checked, as well as Posts. 
3. Add and activate the WordPress.com Editing Toolkit plugin.
4. Add a new page, and add a Blog Posts block.
5. In the page editor, you should see some unformatted sharing / like button output. 
6. On the page preview / frontend, you should not see the sharing buttons / likes (if testing with 11.2-beta).
7. From Settings > Sharing, disable 'Front page, Archive Pages, and Search results' and the page editor should go back to it's normal view (no sharing / like button output).

It looks like this might be the case on all themes - all that I've tested so far display this behavior:  Twenty Twenty One, Astra, BlockBase, Twenty Twenty Two, Twenty Twenty, Twenty Nineteen.

This was reported internally here: p8oabR-Uj-p2#comment-6512

### A clear and concise description of what you expected to happen.

No sharing / like button output should show in the page editor at all.

### What actually happened

Sharing / like button output is showing in the page editor.

### Browser

Google Chrome/Chromium

### Other information

The output in the editor looks like this:
<img width="581" alt="Screenshot 2022-07-29 at 12 19 06" src="https://user-images.githubusercontent.com/16754605/181748353-101facea-1a02-4c16-8431-8446b97ca2a0.png">

There are two other related issues at the moment:
- https://github.com/Automattic/jetpack/issues/24254
- https://github.com/Automattic/jetpack/issues/7361

### Platform (Simple, Atomic, or both?)

Self-hosted

### Reproducibility

Consistent

### Severity

Most (> 50%)

**Comment by jeherve** [comment]
cc @wojtekn and @danielbachhuber, who worked on #24896

**Comment by wojtekn** [comment]
@coder-karen I've checked the provided steps on the local Jetpack site, and I reproduced the issue. Then I reverted https://github.com/Automattic/jetpack/issues/25329, and I still reproduce the issue in the editor.

It seems an editor issue was in place, and in related PR we fixed the frontend part but missed the issue in the backend. Is it what you observed too?

I'm happy to help with checking this though.
